### PR TITLE
Document Qserv cluster installation

### PR DESCRIPTION
**Pull request by jbecla**

### CODE REVIEW

File: admin/bin/qserv-configure.py [hunk 1]
```diff
@@ -85,10 +85,11 @@ def __init__(self):
...
                    "execution\n"
                    "    data for a given Qserv instance.\n"
-                   "  - Create a data directory (QSERV_DATA_DIR) which contains MySQL/Qserv/Zookeeper data.\n"
-                   "  - Use templates and meta-config file parameters (see QSERV_RUN_DIR/qserv-meta.conf) to"
+                   "  - Create a data directory (QSERV_DATA_DIR) which contains MySQL/Qserv/Zookeeper data,\n"
+                   "QSERV_RUN_DIR/var/lib symlink to QSERV_DATA_DIR if different"
+                   "  - Use templates and meta-config file parameters (see QSERV_RUN_DIR/qserv-meta.conf) to "
                    "generate\n"
-                   "    Qserv configuration files and databases.\n"
+                   "    Qserv configuration files and initializes databases.\n"
```
Comment (andy-slac) [inline admin/bin/qserv-configure.py]: `initialize`

File: admin/bin/qserv-configure.py [hunk 2]
```diff
@@ -85,10 +85,11 @@ def __init__(self):
                    "  - Create an execution directory (QSERV_RUN_DIR) which contains configuration and"
                    "execution\n"
                    "    data for a given Qserv instance.\n"
-                   "  - Create a data directory (QSERV_DATA_DIR) which contains MySQL/Qserv/Zookeeper data.\n"
-                   "  - Use templates and meta-config file parameters (see QSERV_RUN_DIR/qserv-meta.conf) to"
+                   "  - Create a data directory (QSERV_DATA_DIR) which contains MySQL/Qserv/Zookeeper data,\n"
+                   "QSERV_RUN_DIR/var/lib symlink to QSERV_DATA_DIR if different"
```
Comment (andy-slac) [inline admin/bin/qserv-configure.py]: Please indent this like and add newline at the end.
You should have run qserv-configure from your branch, you'd see all these formatting issues.

File: admin/bin/qserv-configure.py [hunk 3]
```diff
@@ -85,10 +85,11 @@ def __init__(self):
                    "  - Create an execution directory (QSERV_RUN_DIR) which contains configuration and"
                    "execution\n"
```
Comment (andy-slac) [inline admin/bin/qserv-configure.py]: space needed between "and" and "execution"

File: admin/bin/qserv-configure.py [hunk 4]
```diff
@@ -111,22 +112,20 @@ def __init__(self):
...
-                group = db_component_group
-            elif step_name in configure.NODB_COMPONENTS:
-                group = nodb_component_group
-            else:
+            # complex steps are removed from options
+            elif step_name in configure.ETC + configure.CLIENT:
                 group = config_group
+            else:
+                continue
+
```
Comment (andy-slac) [inline admin/bin/qserv-configure.py]: I think `-R` option and `-D` option logically belong to the same group.

File: doc/source/admin/configuration.rst [hunk 5]
```diff
@@ -2,12 +2,16 @@
...
 
+This documentation presents Qserv configuration tool, designed to simplify and automate Qserv installation process.
+
 Goals
 =====
 
 Qserv installation procedure consists of next steps :
 
-- download, build and install Qserv using eups,
+.. _eups https://github.com/RobertLuptonTheGood/eups
```
Comment (andy-slac) [inline doc/source/admin/configuration.rst]: need colon after `_eups`

File: doc/source/admin/configuration.rst [hunk 6]
```diff
@@ -17,48 +21,32 @@ Approach
 ========
 
 eups install directory must only contains immutable data like, for example, binaries.
-That's why Qserv configuration tool create a separate directory which contains :
+That's why Qserv configuration tool create two directories:
```
Comment (andy-slac) [inline doc/source/admin/configuration.rst]: creates

File: doc/source/admin/configuration.rst [hunk 7]
```diff
@@ -17,48 +21,32 @@ Approach
...
+
+- QSERV_RUN_DIR which contains configuration (file and data) and execution informations (pid files, log files)
+- QSERV_DATA_DIR which contains business data (i.e. sky data),
```
Comment (andy-slac) [inline doc/source/admin/configuration.rst]: Is there a better word than business?

File: doc/source/admin/configuration.rst [hunk 8]
```diff
@@ -17,48 +21,32 @@ Approach
...
 ========
 
-- creates services configuration files in QSERV_RUN_DIR/etc/qserv.conf using a meta-configuration file in QSERV_RUN_DIR/qserv.conf, 
-- runs mysql, scisql and xrootd configuration scripts,
-- creates client configuration,
-- default procedure doesn't remove QSERV_RUN_DIR but only update it,
-- '-all' option allow to create QSERV_RUN_DIR from scratch.
+- Use meta-configuration file QSERV_RUN_DIR/qserv-meta.conf, and templates in QSERV_DIR/cfg/templates to generate all configuration
+- Run configuration scripts for Qserv services when required (i.e. for mysql, scisql and xrootd)
+- Protect existing data, under conditions
```
Comment (andy-slac) [inline doc/source/admin/configuration.rst]: What does it mean "under conditions"?

File: doc/source/admin/configuration.rst [hunk 9]
```diff
@@ -17,48 +21,32 @@ Approach
...
+- Create client configuration in ~/.lsst
 
 Tickets completed
 =================
 
 - https://jira.lsstcorp.org/browse/DM-622
 - https://jira.lsstcorp.org/browse/DM-930
+- https://jira.lsstcorp.org/browse/DM-595
+- https://jira.lsstcorp.org/browse/DM-595
```
Comment (andy-slac) [inline doc/source/admin/configuration.rst]: Duplicate line

File: doc/source/HOW-TO/cluster-deployment.rst [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+Pre-requisites
+++++++++++++++
+
+* Install LSST stack and Qserv dependencies as described here :ref:`quick-start-devel-pre-requisites`.
+* Open write access for each developer to Qserv install path:
+
+  .. code-block:: bash
+
+   # Update acl
+   # TODO: test it
```
Comment (andy-slac) [inline doc/source/HOW-TO/cluster-deployment.rst]: Will you test it?

File: doc/source/HOW-TO/cluster-deployment.rst [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+   setfacl --recursive --modify g::rwx /path/to/lsst/stack/Linux64/qserv
+
+Build and tag
++++++++++++++
+
+And then each developer can build and install in LSST stack its own Qserv
```
Comment (andy-slac) [inline doc/source/HOW-TO/cluster-deployment.rst]: "And then" is not a good start for a new section.

File: doc/source/HOW-TO/cluster-deployment.rst [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+=====================
+
+Adapt this example: 
+
+* Script for installing a cluster node :download:`install-node.sh <../../../admin/tools/cluster/cc-in2p3/install-node.sh>`. In this example, a shared file system is used to retrieve binaries on all nodes.
+* Aliases for parallel management via ssh :download:`parallel-ssh-command.aliases <../../../admin/tools/cluster/cc-in2p3/parallel-ssh-command.aliases>`.
+
+  .. code-block:: bash
+
+   source ~/src/qserv/admin/tools/cluster/cc-in2p3/parallel-ssh-command.aliases
```
Comment (andy-slac) [inline doc/source/HOW-TO/cluster-deployment.rst]: Is this the file that one is supposed to modify first?

File: admin/bin/qserv-configure.py (hunk 1 above)
Comment (fjammes) [inline admin/bin/qserv-configure.py]: ok

File: admin/bin/qserv-configure.py (hunk 2 above)
Comment (fjammes) [inline admin/bin/qserv-configure.py]: ok

File: admin/bin/qserv-configure.py (hunk 4 above)
Comment (fjammes) [inline admin/bin/qserv-configure.py]: -R option must be used on each step if needed, -D option should only be used at initialization step. That's why they're not in the same group.

File: doc/source/admin/configuration.rst (hunk 6 above)
Comment (fjammes) [inline doc/source/admin/configuration.rst]: ok thanks :+1:

File: doc/source/admin/configuration.rst (hunk 7 above)
Comment (fjammes) [inline doc/source/admin/configuration.rst]: I replace with 'scientific' :+1:

File: doc/source/admin/configuration.rst (hunk 8 above)
Comment (fjammes) [inline doc/source/admin/configuration.rst]: I replace with "if located outside of configuration directory"

File: doc/source/admin/configuration.rst (hunk 9 above)
Comment (fjammes) [inline doc/source/admin/configuration.rst]: :+1:

File: doc/source/HOW-TO/cluster-deployment.rst (hunk 10 above)
Comment (fjammes) [inline doc/source/HOW-TO/cluster-deployment.rst]: I need somebody else with a account on the cluster to confirm me it works. I removed the comment.

File: doc/source/HOW-TO/cluster-deployment.rst (hunk 11 above)
Comment (fjammes) [inline doc/source/HOW-TO/cluster-deployment.rst]: :+1:

File: doc/source/HOW-TO/cluster-deployment.rst (hunk 12 above)
Comment (fjammes) [inline doc/source/HOW-TO/cluster-deployment.rst]: No, it should work out of the box.

**Comment by fjammes** [comment]
Can I merge now please?

**Comment by andy-slac** [comment]
No objections.

**Comment by fritzm** [comment]
Merged May 19, 2015 -- SHA bca16d1

**Pull request closed without merging**